### PR TITLE
Add to view custom attribute close #89

### LINKED ISSUE #89: Feature Request - custom attribute method
If I define methods in my `**view.ex` file, I want to include them in the attributes hash.  Once could imagine combining `first + last` or some type of conditional logic for a specific field at the view layer.

I was wondering your thoughts were on this and if it is something you want me to take a stab at...

```
defmodule MyApp.ArticleSerializer do

  def fields, do: [:body, :excerpt]

  def excerpt(article, _conn) do
    [first | _ ] = String.split(article.body, ".")
    first
  end
end
```

### PR DESCRIPTION
**Pull request by snewcomer**
Ref #89

### CODE REVIEW

File: lib/jsonapi/view.ex [hunk 1]
```diff
@@ -91,7 +91,19 @@ defmodule JSONAPI.View do
...
+        Enum.reduce(visible_fields, %{}, fn field, intermediate_map ->
+          try do
+            merge(intermediate_map, field, apply(__MODULE__, field, [data, conn]))
+          rescue
+            _e -> merge(intermediate_map, field, Map.get(data, field))
+          end
+        end)
+      end
+
+      defp merge(intermediate_map, field, value) do
```
Comment (doomspork) [inline lib/jsonapi/view.ex]: What does `merge/3` offer us over just using `Map.put/3` directly?

File: lib/jsonapi/view.ex [hunk 2]
```diff
@@ -91,7 +91,19 @@ defmodule JSONAPI.View do
       end
 
       def attributes(data, conn) do
-        visible_fields(data)
+        visible_fields = fields() -- hidden()
+
+        Enum.reduce(visible_fields, %{}, fn field, intermediate_map ->
```
Comment (doomspork) [inline lib/jsonapi/view.ex]: Why `reduce/3` vs `map/2`?

File: lib/jsonapi/view.ex [hunk 3]
```diff
@@ -91,7 +91,19 @@ defmodule JSONAPI.View do
...
+          try do
+            merge(intermediate_map, field, apply(__MODULE__, field, [data, conn]))
```
Comment (doomspork) [inline lib/jsonapi/view.ex]: Can we do something like this to DRY things up a bit?

```elixir
value =
  try do
    apply(__MODULE__, field, [data, conn])
  rescue
    _e -> Map.get(data, field)
  end

Map.put(intermediate_map, field, value)
```

File: lib/jsonapi/view.ex (hunk 2 above)
Comment (snewcomer) [reply lib/jsonapi/view.ex]: This was to ensure the return type was a Map, which is what `Map.take` was doing before.

File: lib/jsonapi/view.ex (hunk 3 above)
Comment (snewcomer) [reply lib/jsonapi/view.ex]: ah much better :100:

**Comment by doomspork** [comment]
Thank you @snewcomer!  I want to review #87 and then we can cut a new release 😁

**Pull request merged**